src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -525,6 +527,19 @@ private boolean isContainVerifiableType(DetailAST modifiers) {
         return result;
     }
 
+    /**
+     * Checks a record component definition. Record components are private
+     * final fields, so we check code in a similar manner as we do for
+     * private final class members.
```

Comment:
```
* final fields, so we do not consider memberModifiers field.
```

Reply:
Done

Reply:
I will surely do it when I push changes again.

Reply:
Done